Main/src/main/java/com/main/acad/controller/Controller.java

```diff
@@ -1,37 +1,53 @@
 package com.main.acad.controller;
 
 import com.main.acad.annotation.MappingMethod;
-import com.main.acad.dao.ChapterDao;
-import com.main.acad.dao.Dao;
 import com.main.acad.entity.Chapter;
-import com.main.acad.serializator.MyJsonSerializer;
+import com.main.acad.serializator.JsonSerializatorImplementation;
+import com.main.acad.service.StrategySimulationServiceImplementation;
 
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.io.IOException;
-import java.io.PrintWriter;
+import java.sql.SQLException;
 import java.util.List;
+import java.util.logging.Logger;
 
 public class Controller {
+    private static final Logger logger = Logger.getLogger(Controller.class.getName());
+
+    StrategySimulationServiceImplementation strategySimulationServiceImplementation = StrategySimulationServiceImplementation.getInstance();
+
     @MappingMethod(url = "api/chapter")
-    public HttpServletResponse returnChapters(HttpServletRequest request, HttpServletResponse response)
-            throws ClassNotFoundException, IllegalAccessException, IOException {
-        Dao dao = new ChapterDao();
-        List chaptersList = dao.listChapters();
-        MyJsonSerializer m = new MyJsonSerializer();
-        String result = m.write(chaptersList);
-        response.getWriter().write(result);
-        return response;
+    public void returnChapters(HttpServletRequest request, HttpServletResponse response) {
+        try {
+            List chaptersList = strategySimulationServiceImplementation.getlistChapters();
+            JsonSerializatorImplementation jsonSerializator = new JsonSerializatorImplementation();
+            String result = jsonSerializator.write(chaptersList);
+            response.getWriter().write(result);
+        } catch (IllegalAccessException | IOException e) {
+            logger.info("An error occurred in the Controller in the returnChapters method");
```

Comment:
it is a bad idea to catch exception and do nothing except logging. Please rethrow one of unchecked exception (it will be really hard to find out what's going on if something will not work as expected)

Reply:
exception handling too

Reply:
Can you write throw new {some runtime exception}(e.message) in the catch section instead of printing stacktrace?